bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java

```diff
@@ -307,26 +318,44 @@ protected void doConnectedRun() throws IOException, ApiException {
         }
     };
 
+    private final Runnable scenePollingRunnable = new PollingRunnable() {
+        @Override
+        protected void doConnectedRun() throws IOException, ApiException {
+            Map<String, String> groupNames = lastGroupStates.entrySet().parallelStream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().getName()));
+            List<StateOption> stateOptions = hueBridge.getScenes().parallelStream()//
+                    .map(scene -> scene.toStateOption(groupNames))//
+                    .collect(Collectors.toList());
```

Comment:
Please consider my proposal to compute the room when not returned by the API.

Reply:
The problem: according to the documentation, there are two kinds of scenes. For `GroupScene`s, the group ID is returned. The other kind are `LightScene`s. For those scenes, the api returns only the contained lights. As lights can be contained in multiple groups, a `LightScene`s group might not be uniquely determined

Reply:
That's why I set the group only if all lights are in the same room.
IN my case (bridge v1), the group is never returned by the API. Determining it allows to display it in the options and so to differenciate different scenes which have the same name for all rooms.